doc/user/content/ingest-data/sql-server/self-hosted.md

```diff
@@ -276,6 +279,116 @@ available(also for PostgreSQL)."
 
 {{% sql-server-direct/next-steps %}}
 
+## High Availability
+
+To make your SQL Server source resilient to database failovers, configure
+Materialize to connect through a SQL Server [Always On Availability Group (AG)
+listener](https://learn.microsoft.com/en-us/sql/database-engine/availability-groups/windows/listeners-client-connectivity-application-failover).
+When a failover occurs, SQL Server drops the existing connection and routes new
+connections to the new primary replica transparently.
+
+{{< warning >}}
+**Availability modes and data consistency:** SQL Server AGs support two
+[availability modes](https://learn.microsoft.com/en-us/sql/database-engine/availability-groups/windows/overview-of-always-on-availability-groups-sql-server?view=sql-server-ver17#availability-modes):
+synchronous-commit and asynchronous-commit.
+
+With **asynchronous-commit mode**, transactions commit on the primary before
+being sent to secondaries. If the primary fails before replicating recent
+transactions, those changes will be lost and **Materialize will not ingest
+them**. For guaranteed data consistency, use **synchronous-commit mode**.
+
+For additional best practices on configuring CDC with availability groups, see
+[Microsoft's documentation on replication agents with availability groups](https://learn.microsoft.com/en-us/sql/database-engine/availability-groups/windows/replicate-track-change-data-capture-always-on-availability?view=sql-server-ver17#general-changes-to-replication-agents-to-support-availability-groups).
+{{< /warning >}}
+
+#### Prerequisites
+
+Before connecting Materialize to an AG, ensure:
+
+1. **Your AG listener is configured and accessible.** Materialize must connect
```

Comment:
Should we mention the synchronous-commit mode here? Something like (spitballing):

1. **Your AG listener is configured (e.g., uses synchronous-commit mode and is accessible).** ....